### PR TITLE
Initial commit of FusionDirectory

### PR DESCRIPTION
**Pull request by dasJ**

### CODE REVIEW

File: tasks/install_deb.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+- name: Update apt cache
+  apt:
+    update_cache: yes
+  changed_when: false
```
Comment (Mr-Pi) [inline tasks/install_deb.yml:5]: This is done by a common role and should not be done here.

File: vars/main.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+local_download_dir: "{{ global_cache_dir | mandatory }}/fusiondirectory"
+local_fd_dir: "{{ local_download_dir }}/fusiondirectory-{{ fusiondirectory_version | mandatory }}"
+local_plugins_dir: "{{ local_download_dir }}/fusiondirectory-plugins-{{ fusiondirectory_version | mandatory }}"
+fd_download_url: "https://repos.fusiondirectory.org/sources/{{ fusiondirectory_major_version }}/fusiondirectory/fusiondirectory-{{ fusiondirectory_version }}.tar.gz"
+plugins_download_url: "https://repos.fusiondirectory.org/sources/{{ fusiondirectory_major_version }}/fusiondirectory/fusiondirectory-plugins-{{ fusiondirectory_version }}.tar.gz"
```
Comment (Mr-Pi) [inline vars/main.yml:6]: Please prefix variables

File: tasks/plugins.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+- name: Create plugin dir
+  file:
+    path: "{{ local_download_dir }}/fd-plugins"
+    state: directory
+  become: no
+  run_once: true
+  delegate_to: localhost
+
+- name: Copy plugins
+  command: rsync -varuEAXH --delete "{{ item }}/" "{{ local_download_dir }}/fd-plugins/{{ item | basename }}/"
```
Comment (Mr-Pi) [inline tasks/plugins.yml:22]: please add rsync to packages to be installed.

**Review by Mr-Pi — DISMISSED** [review]

File: tasks/install_deb.yml (hunk 1 above)
Comment (SlothOfAnarchy) [reply tasks/install_deb.yml:5]: Not everyone will use or execute the common role

File: README.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+| `global_cache_dir`                 | :heavy_check_mark: | Path to a directory on the control machine where FusionDirectory is downloaded and extracted to |
+| `fusiondirectory_major_version`    | `1.0`              | Major version of FusionDirectory, this is required in the download URL                          |
+| `fusiondirectory_version`          | `1.1.1`            | FusionDirectory version to download and extract                                                 |
+| `fusiondirectory_http_group`       | `www-data`         | Group under which the web server runs. This group is allowed to read FusionDirectory's files    |
+| `fusiondirectory_force_ssl`        | `false`            | Force usage of SSL                                                                              |
+| `fusiondirectory_ignore_acl`       |                    | Specification of a DN that ignores all FusionDirectory ACLs                                     |
+| `fusiondirectory_logging`          | `true`             | Enable logging                                                                                  |
```
Comment (SlothOfAnarchy) [inline README.md:39]: Where are the logs placed? Is it configurable?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+Only the PHP application is downloaded (on the control machine), extracted and configured.
+No data is written to your LDAP server, this has to be done by another role.
+Also, you need a PHP-capable webserver which serves `/usr/share/webapps/fusiondirectory`.
+
+The schemas are not touched, [fusiondirectory_schemas](https://github.com/stuvusIT/fusiondirectory_schemas) is used for this.
+Your LDAP server needs to have some schemas loaded, see the Requirements section for that.
+
+## Requirements
+
+A dpkg- or pacman-based Linux distribution.
```
Comment (SlothOfAnarchy) [inline README.md:13]: There are no pacman tasks yet.

File: tasks/plugins.yml [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+- name: Copy plugins
+  copy:
+    src: "{{ local_download_dir }}/fd-plugins.tar.gz"
+    dest: "{{ plugin_tmpfile.path }}/"
+  changed_when: false
+
+- name: Install plugins
+  shell: "echo '{{ plugin_tmpfile.path }}/fd-plugins.tar.gz' | fusiondirectory-setup --set-fd_home=/usr/share/webapps/fusiondirectory --install-plugins --update-locales --update-cache"
+  changed_when: false
```
Comment (SlothOfAnarchy) [inline tasks/plugins.yml:53]: can this setup command be executed multiple times without changes or sideeffects?

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

File: tasks/install_deb.yml (hunk 1 above)
Comment (dasJ) [reply tasks/install_deb.yml:5]: @Mr-Pi Are you serious???

File: vars/main.yml (hunk 2 above)
Comment (dasJ) [reply vars/main.yml:6]: No. They are role-internal and cannot be changed

File: README.md (hunk 5 above)
Comment (dasJ) [reply README.md:13]: Damn

File: README.md (hunk 4 above)
Comment (dasJ) [reply README.md:39]: Syslog, so no

File: tasks/plugins.yml (hunk 6 above)
Comment (dasJ) [reply tasks/plugins.yml:53]: Yes, I tried

File: tasks/plugins.yml (hunk 3 above)
Comment (dasJ) [reply tasks/plugins.yml:22]: Done

**Comment by dasJ** [comment]
@Mr-Pi @SlothOfAnarchy Pls rereview

**Review by SlothOfAnarchy — APPROVED** [review]

**Comment by SlothOfAnarchy** [comment]
@Mr-Pi Your requested changes have been implemented or discussed. Can you approve and merge this?
Thanks!

**Pull request merged**